tests/e2e_test.go

```diff
@@ -359,3 +374,117 @@ func TestWatch(t *testing.T) {
 	}
 	fmt.Println(fi[0].Name())
 }
+
+func TestCreateLogEntrySignature(t *testing.T) {
+	// Create a random payload and sign it
+	ctx := context.Background()
+	payload := []byte("payload")
+	s, err := signer.NewMemory()
+	if err != nil {
+		t.Fatal(err)
+	}
+	signature, _, err := s.Sign(ctx, payload)
+	if err != nil {
+		t.Fatal(err)
+	}
+	pubkey, err := s.PublicKey(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	b, err := x509.MarshalPKIXPublicKey(pubkey)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	pemBytes := pem.EncodeToMemory(&pem.Block{
+		Type:  "PUBLIC KEY",
+		Bytes: b,
+	})
+
+	// submit our newly signed payload to rekor
+	rekorClient, err := app.GetRekorClient("http://localhost:3000")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	re := rekord.V001Entry{
+		RekordObj: models.RekordV001Schema{
+			Data: &models.RekordV001SchemaData{
+				Content: strfmt.Base64(payload),
+			},
+			Signature: &models.RekordV001SchemaSignature{
+				Content: strfmt.Base64(signature),
+				Format:  models.RekordV001SchemaSignatureFormatX509,
+				PublicKey: &models.RekordV001SchemaSignaturePublicKey{
+					Content: strfmt.Base64(pemBytes),
+				},
+			},
+		},
+	}
+
+	returnVal := models.Rekord{
+		APIVersion: swag.String(re.APIVersion()),
+		Spec:       re.RekordObj,
+	}
+	params := entries.NewCreateLogEntryParams()
+	params.SetProposedEntry(&returnVal)
+	resp, err := rekorClient.Entries.CreateLogEntry(params)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// unmarshal the response
+	le := resp.GetPayload()
+	if len(le) != 1 {
```

Comment:
we should make a helper for this at some point (getting the one field out), it's so annoying to get right and we do it all over the place

Reply:
I added it as a helper func for testing, but if we're doing it in code I could move it to a general util package or something